Archive/Imo/Imo2020Q6.lean

```diff
@@ -0,0 +1,332 @@
+  intro y hy
+  grw [← abs_signedDist_le_dist (b -ᵥ a)]
+  congr!
+  rw [AffineSubspace.mem_mk', LinearMap.mem_ker, innerₛₗ_apply] at hy
+  rw [AffineMap.lineMap_apply, vsub_vadd_eq_vsub_sub y, inner_sub_right, sub_eq_zero] at hy
+  rw [sub_eq_iff_eq_add', signedDist_triangle]
+  rw [signedDist_apply_apply, NormedSpace.normalize, real_inner_smul_left, hy]
+  rw [inner_smul_right, real_inner_self_eq_norm_sq]
+  field_simp
+
+theorem card_le_of_separated {ι : Type*} (S : Finset ι) (f : ι → ℝ) {ε a b : ℝ} (hε : 0 < ε)
+    (hab : a ≤ b) (h_sep : (S : Set ι).Pairwise fun x y => ε ≤ dist (f x) (f y))
+    (hbound : ∀ x ∈ S, f x ∈ Set.Icc a b) : #S ≤ ⌊(b - a) / ε + 1⌋ := by
+  suffices h : #S ≤ #(Icc 0 ⌊(b - a) / ε⌋) by
+    have : 0 ≤ ⌊(b - a) / ε⌋ + 1 := by
+      have := sub_nonneg_of_le hab
+      positivity
+    simpa [this] using h
+  apply Finset.card_le_card_of_injOn fun x => ⌊(f x - a) / ε⌋
+  · intro x hx
+    rw [coe_Icc, Set.mem_Icc]; constructor
+    · rw [Int.floor_nonneg]
+      refine div_nonneg ?_ hε.le
+      rw [sub_nonneg]
+      exact (hbound x hx).1
+    · apply Int.floor_le_floor
+      rw [div_le_div_iff_of_pos_right hε, sub_le_sub_iff_right]
+      exact (hbound x hx).2
+  · intro x hx y hy h
+    apply Int.abs_sub_lt_one_of_floor_eq_floor at h
+    field_simp at h
+    rw [sub_sub_sub_cancel_right, abs_div, abs_eq_self.mpr hε.le, div_lt_one hε] at h
+    contrapose! h
+    exact h_sep hx hy h
+
+/--
+In a strip of width `1/2`, if the points have pairwise distance at least `1`,
+then we can bound the number of points.
+-/
+theorem card_le_of_separated_in_strip (eqv : P ≃ᵃⁱ[ℝ] EuclideanSpace ℝ (Fin 2)) (S : Finset P)
+    (h_sep : (S.filter (eqv · 0 ∈ Set.Ioo 0 (1 / 2)) : Set P).Pairwise fun x y => 1 ≤ dist x y)
+    {N : ℝ} (hN : 1 ≤ N) (h_bound : ∀ x ∈ S.filter (eqv · 0 ∈ Set.Ioo 0 (1 / 2)), |eqv x 1| ≤ N) :
+    #(S.filter (eqv · 0 ∈ Set.Ioo 0 (1/2))) ≤ N*6-1 := by
+  suffices h : #(S.filter (eqv · 0 ∈ Set.Ioo 0 (1/2))) ≤ ⌊(N - (-N)) / (1/2) + 1⌋ by
+    rw [Int.le_floor, Int.cast_natCast] at h
+    linarith only [h, hN]
+  refine card_le_of_separated (S.filter (eqv · 0 ∈ Set.Ioo 0 (1/2))) (eqv · 1)
+    (by norm_num) (by linarith only [hN]) ?_
+    (fun x hx => ⟨neg_le_of_abs_le (h_bound x hx), le_of_abs_le (h_bound x hx)⟩)
+  intro x hx y hy h_ne
+  specialize h_sep hx hy h_ne
+  dsimp only
+  have := EuclideanSpace.dist_eq (eqv x) (eqv y)
+  simp at this
+  rw [this] at h_sep
+  rw [Real.one_le_sqrt] at h_sep
+  rw [← sq_le_sq₀ (by positivity) (by positivity)]
+  suffices (dist (eqv x 0) (eqv y 0))^2 ≤ (1/2)^2 by
+    linarith only [this, h_sep]
+  rw [sq_le_sq₀ (by positivity) (by positivity), Real.dist_eq, abs_le]
+  rw [coe_filter] at hx hy
+  rcases hx with ⟨_, xl, xr⟩
+  rcases hy with ⟨_, yl, yr⟩
+  constructor
+  · linear_combination xl + yr
+  · linear_combination xr + yl
+
+
+variable [Fact (Module.finrank ℝ V = 2)]
+
+theorem result : ∃ c : ℝ, 0 < c ∧ ∀ {n : ℕ}, 1 < n → ∀ {S : Finset P}, #S = n →
```

Comment:
This puts `result` in the root namespace. In other IMO files in `Archive/`, we either have a namespace like `namespace Imo2020Q6` or we name the result something unique like `theorem imo2020q6`.